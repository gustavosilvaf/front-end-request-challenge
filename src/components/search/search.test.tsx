import { render } from "@testing-library/react";
import { fireEvent } from "@testing-library/dom";
import { Search } from ".";

const mockSetValue = jest.fn();
const mockSubmit = jest.fn((e) => e.preventDefault());
const mockOptions = ["First shirt, second shirt"];

describe("Search tests", () => {
  it("Should render corretly", () => {
    const { asFragment } = renderComponent();
    expect(asFragment).toMatchSnapshot();
  });

  it("Should render input and button", () => {
    const { getByPlaceholderText, getByRole } = renderComponent();

    expect(getByRole("button")).toBeInTheDocument();
    expect(getByPlaceholderText("Search a product")).toBeInTheDocument();
  });

  it("Should call setValue when change input value", () => {
    const { getByPlaceholderText } = renderComponent();

    const input = getByPlaceholderText("Search a product");

    fireEvent.change(input, { target: { value: "Shirt" } });

    expect(mockSetValue).toHaveBeenCalledWith("Shirt");
  });

  it("Should submit form when button was clicked", () => {
    const { getByRole } = renderComponent();

    const button = getByRole("button");

    fireEvent.click(button);

    expect(mockSubmit).toHaveBeenCalled();
  });
});

const renderComponent = () =>
  render(
    <Search
      handleSearch={mockSubmit}
      options={mockOptions}
      value={""}
      setValue={mockSetValue}
    />
  );
