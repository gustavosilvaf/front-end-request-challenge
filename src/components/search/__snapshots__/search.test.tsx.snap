// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests Should render corretly 1`] = `[Function]`;
